File: flutter_zoom_videosdk-2.4.0/ios/Classes/FlutterZoomVideoSdkPhoneHelper.m
#import "FlutterZoomVideoSdkPhoneHelper.h"
#import "FlutterZoomVideoSdkPhoneSupportCountryInfo.h"
#import "JSONConvert.h"
#import "FlutterZoomVideoSdkSessionDialInNumberInfo.h"

@implementation FlutterZoomVideoSdkPhoneHelper

- (ZoomVideoSDKPhoneHelper *) getPhoneHelper {
    ZoomVideoSDKPhoneHelper* phoneHelper = nil;
    @try {
        phoneHelper = [[ZoomVideoSDK shareInstance] getPhoneHelper];
        if (phoneHelper == nil) {
            NSException *e = [NSException exceptionWithName:@"NoPhoneHelperFound" reason:@"No Phone Helper Found" userInfo:nil];
            @throw e;
        }
    } @catch (NSException *e) {
       NSLog(@"%@ - %@", e.name, e.reason);
    }
    return phoneHelper;
}

-(void) isSupportPhoneFeature: (FlutterResult) result {
    if ([[self getPhoneHelper] isSupportPhoneFeature]) {
        result(@YES);
    } else {
        result(@NO);
    }
}

-(void) getSupportCountryInfo: (FlutterResult) result {
    ZoomVideoSDKPhoneHelper* phoneHelper = [[ZoomVideoSDK shareInstance] getPhoneHelper];

    if (phoneHelper == nil) {
        result(@"phoneHelper_getSupportCountryInfo: No phone helper found");
    }

    result([FlutterZoomVideoSdkPhoneSupportCountryInfo mapPhoneSupportCountryInfo:[phoneHelper getSupportCountryInfo]]);
}

-(void) inviteByPhone: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    ZoomVideoSDKPhoneHelper* phoneHelper = [[ZoomVideoSDK shareInstance] getPhoneHelper];

    NSLog(@"countryCode:%@ , phoneNumber:%@ , name:%@", call.arguments[@"countryCode"], call.arguments[@"phoneNumber"], call.arguments[@"name"]);

    if (phoneHelper.isSupportPhoneFeature == YES) {
        dispatch_async(dispatch_get_main_queue(), ^{
            result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([phoneHelper inviteByPhone: call.arguments[@"countryCode"] phoneNumber: call.arguments[@"phoneNumber"] name: call.arguments[@"name"]])]);
        });
    }
}

-(void) cancelInviteByPhone: (FlutterResult) result {
    ZoomVideoSDKPhoneHelper* phoneHelper = [[ZoomVideoSDK shareInstance] getPhoneHelper];

    if (phoneHelper.isSupportPhoneFeature == YES) {
        dispatch_async(dispatch_get_main_queue(), ^{
            result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([phoneHelper cancelInviteByPhone])]);
        });
    }
}

-(void) getInviteByPhoneStatus: (FlutterResult) result {

    result([[JSONConvert ZoomVideoSDKPhoneStatusValuesReversed] objectForKey: @([[self getPhoneHelper] getInviteByPhoneStatus])]);
}

-(void) getSessionDialInNumbers: (FlutterResult) result {
    ZoomVideoSDKPhoneHelper* phoneHelper = [[ZoomVideoSDK shareInstance] getPhoneHelper];
    result([FlutterZoomVideoSdkSessionDialInNumberInfo mapSessionDialInNumberInfoArray:[phoneHelper getSessionDialInNumbers]]);
}

@end

File: flutter_zoom_videosdk-2.4.0/android/src/main/java/com/flutterzoom/videosdk/FlutterZoomVideoSdkPhoneHelper.java
package com.flutterzoom.videosdk;

import android.app.Activity;

import androidx.annotation.NonNull;

import com.flutterzoom.videosdk.convert.FlutterZoomVideoSdkErrors;
import com.flutterzoom.videosdk.convert.FlutterZoomVideoSdkPhoneStatus;

import java.util.List;
import java.util.Map;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import us.zoom.sdk.ZoomVideoSDK;
import us.zoom.sdk.ZoomVideoSDKErrors;
import us.zoom.sdk.ZoomVideoSDKPhoneHelper;
import us.zoom.sdk.ZoomVideoSDKPhoneSupportCountryInfo;
import us.zoom.sdk.ZoomVideoSDKSessionDialInNumberInfo;

public class FlutterZoomVideoSdkPhoneHelper {

    private Activity activity;

    FlutterZoomVideoSdkPhoneHelper(Activity activity) {
        this.activity = activity;
    }

    private ZoomVideoSDKPhoneHelper getPhoneHelper() {
        ZoomVideoSDKPhoneHelper phoneHelper = null;

        try {
            phoneHelper = ZoomVideoSDK.getInstance().getPhoneHelper();
            if (phoneHelper  == null) {
                throw new Exception("No Phone Helper Found");
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return phoneHelper;
    }

    public void cancelInviteByPhone(@NonNull MethodChannel.Result result) {
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getPhoneHelper().cancelInviteByPhone()));
            }
        });
    }

    public void getInviteByPhoneStatus(@NonNull MethodChannel.Result result) {
        result.success(FlutterZoomVideoSdkPhoneStatus.valueOf(getPhoneHelper().getInviteByPhoneStatus()));
    }

    public void getSupportCountryInfo(@NonNull MethodChannel.Result result) {

        List<ZoomVideoSDKPhoneSupportCountryInfo> countryList = getPhoneHelper().getSupportCountryInfo();

        if (countryList == null) {
            result.error("ZoomVideoSdkPhoneHelper::getSupportCountryInfo", "Unable to get support country info", null);
        }

        result.success(FlutterZoomVideoSdkPhoneSupportCountryInfo.jsonPhoneSupportCountryInfoArray(countryList));
    }

    public void inviteByPhone(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> params = call.arguments();
        String countryCode = (String) params.get("countryCode");
        String phoneNumber = (String) params.get("phoneNumber");
        String name = (String) params.get("name");
        if (countryCode.isEmpty() || phoneNumber.isEmpty()) {
            result.error(FlutterZoomVideoSdkErrors.valueOf(ZoomVideoSDKErrors.Errors_Invalid_Parameter), "countryCode or phoneNumber is null or empty", null);
        }

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getPhoneHelper().inviteByPhone(countryCode, phoneNumber, name)));
            }
        });
    }

    public void isSupportPhoneFeature(@NonNull MethodChannel.Result result) {
        result.success(getPhoneHelper().isSupportPhoneFeature());
    }

    public void getSessionDialInNumbers(@NonNull MethodChannel.Result result) {
        List<ZoomVideoSDKSessionDialInNumberInfo> sessionDialInNumberList = getPhoneHelper().getSessionDialInNumbers();

        if (sessionDialInNumberList == null) {
            result.error("ZoomVideoSdkPhoneHelper::getSessionDialInNumbers", "there is no session dial-in numbers", null);
        }

        result.success(FlutterZoomVideoSdkSessionDialInNumberInfo.jsonSessionDialInNumberInfoArray(sessionDialInNumberList));
    }

}
